easybuild/easyblocks/m/motioncor2.py

```diff
@@ -0,0 +1,112 @@
+##
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MotionCor2, implemented as an easyblock
+
+@author: Ake Sandgren, (HPC2N, Umea University)
+"""
+
+import os
+import stat
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, mkdir, write_file
+
+
+class EB_MotionCor2(EasyBlock):
+    """
+    Support for installing MotionCor2
+     - creates wrapper that loads the correct version of CUDA before
+     - running the actual binary
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EB_MotionCor2, self).__init__(*args, **kwargs)
+
+        self.cuda_mod_name, self.cuda_name = None, None
+        for dep in self.cfg.dependencies():
+            if dep['name'] == 'CUDA':
+                self.cuda_mod_name = dep['short_mod_name']
+                self.cuda_name = self.cuda_mod_name.rsplit('/', 1)[0]
```

Comment:
@akesandgren Any reason not to use `os.path.dirname`? Easier to understand imho, and the module name is basically a file path, so it makes sense imho...

Reply:
Didn't think of it. For me, treating it as a path is odd in this context, but if you prefer it that way I can change it.

Reply:
I think it would be a lot easier to understand what's going on here, compared to the `rsplit` you're using now.

In fact, it's even wrong:

```
>>> 'foo/CUDA/1.0'.rsplit('/')[0]
'foo'
>>> os.path.dirname('foo/CUDA/1.0')
'foo/CUDA'
```

Reply:
No it isn't wrong, you're just not doing what I am :-)

rsplit('/', 1)

Reply:
Well, that only proves my point that `os.path.dirname` makes more sense. ;)